File: app/src/main/scala/com/trade/figurationService/FigurationService.java
package com.trade.figurationService;

import com.crankuptheamps.client.HAClient;
import com.crankuptheamps.client.Message;
import com.crankuptheamps.client.MessageStream;
import com.trade.amps.AmpsClientUtil;

public class FigurationService {
    public static void main(String[] args) throws Exception{
        HAClient client = AmpsClientUtil.connect("FigurationService");
        System.out.println("Figuration Service connected, waiting for trades...");

        MessageStream ms = client.subscribe("trades.validated").timeout(0);
        for (Message msg: ms){
            String trade = msg.getData();

            //Add price of calculation
            String pricedTrade = trade.replace("}", ",\"price\":150}");
            client.publish("trades.validated", pricedTrade);
            System.out.println("Figured trade: "+pricedTrade);
        }

        client.wait(20000);
        client.close();
    }
}

File: app/src/main/scala/com/trade/figurationService/FigurationService.scala
//package com.trade.figurationService
//
//import com.crankuptheamps.client.{HAClient, Message}
//import com.trade.amps.AmpsClientUtil
//import com.trade.common.Trade
//import play.api.libs.json.Json
//
//object FigurationService extends App {
//
//  val client: HAClient = AmpsClientUtil.connect("FigurationService")
//  println("Figuration Service connected")
//
//  val ms = client.subscribe("trades.validated").timeout(0)
//
//  while (true) {
//    val msg = ms.next()
//    val trade = Json.parse(msg.getData).as[Trade]
//
//    val pricedTrade = trade.copy(price = Some(150.0))
//    val json = Json.stringify(Json.toJson(pricedTrade))
//
//    client.publish("trades.figured", json)
//    println(s"Figured trade: $json")
//  }
//}
//
